src/gc/gc.cpp

```diff
@@ -838,7 +831,7 @@ class t_join
             return TRUE;
         }
 
-        if (Interlocked::Decrement(&join_struct.r_join_lock) != (join_struct.n_threads - 1))
+        if (Interlocked::CompareExchange(&join_struct.r_join_lock, 0, join_struct.n_threads) == 0)
```

Comment:
@Maoni0 In general I was referring to saving the write, reduces MP cache misses which can be become significant.  In this case only a single write should help too.